pipeline_dp/pipeline_operations.py

```diff
@@ -176,8 +188,19 @@ def keys(self, col, stage_name: str):
     def values(self, col, stage_name: typing.Optional[str] = None):
         return (v for k, v in col)
 
-    def sample_fixed_per_key(self, col, n: int, stage_name: str):
-        pass
+    def sample_fixed_per_key(self, col, n: int,
+                             stage_name: typing.Optional[str] = None):
+        def sample_fixed_per_key_generator():
+            for item in self.group_by_key(col):
+                key = item[0]
+                values = item[1]
+                if len(values) > n:
+                    sampled_indices = np.random.choice(range(len(values)), n,
+                                                       replace=False)
+                    values = [values[i] for i in sampled_indices]
+                yield key, values
+
+        return sample_fixed_per_key_generator()
```

Comment:
> I recalled that computations should be lazy, it can be done by using the same patten as in group_by_key
> PipelineDP/pipeline_dp/pipeline_operations.py
> 
> Line 162 in e30b088
> 
> 	 def group_by_key_generator(): 
> Namely, implementation of sampling per key is in a generator and sample_fixed_per_key returns this generator
 
Done